lms/services/assignment.py

```diff
@@ -10,55 +12,110 @@ def __init__(self, db):
         self._db = db
 
     @lru_cache(maxsize=128)
-    def get(self, tool_consumer_instance_guid, resource_link_id):
-        return (
+    def get(
+        self,
+        tool_consumer_instance_guid,
+        resource_link_id=None,
+        ext_lti_assignment_id=None,
+    ):
+        """
+        Get an assignment using using either resource_link_id or ext_lti_assignment_id.
+
+        When a canvas assignment has just been configured it will have
+            resource_link_id=None and ext_lti_assignment_id!=None
+        until it's first lunched.
+        """
+        assignment = (
             self._db.query(Assignment)
-            .filter_by(
-                tool_consumer_instance_guid=tool_consumer_instance_guid,
-                resource_link_id=resource_link_id,
+            .filter(
+                Assignment.tool_consumer_instance_guid == tool_consumer_instance_guid,
+                (
+                    (Assignment.resource_link_id == resource_link_id)
+                    | (Assignment.ext_lti_assignment_id == ext_lti_assignment_id)
+                ),
             )
             .one_or_none()
         )
 
-    def get_document_url(self, tool_consumer_instance_guid, resource_link_id):
+        if not assignment:
+            return None
+
+        if resource_link_id and not assignment.resource_link_id:
+            # First lunch of the assignment, fill the resource_link_id now that we have it.
+            assignment.resource_link_id = resource_link_id
```

Comment:
What about the case where we find an assignment with a matching `ext_lti_assignment_id` and it has a *different* `resource_link_id` than the one we received in the launch params? That should never happen, but maybe we want to make sure that we catch that case and raise an error